### ISSUE TITLE
DataGrid accessibility issue

### ISSUE THREAD

**Issue by SeonyuK** [carbon-design-system/ibm-products#3874]
## What package(s) are you using?

- [X] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

The Datagrid still shows the accessibility violation 

> Is this issue related to a specific component?

Datagrid

> What did you expect to happen? What happened instead? What would you like to
> see changed?

No accessibility test failure 

> What browser are you working in?

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

2.17.0

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> https://codesandbox.io/s/github/carbon-design-system/ibm-products/tree/main/examples/carbon-for-ibm-products/SidePanel

## Additional information

- Screenshots or code
- Notes

Fix any of the following:
      ARIA role separator is not allowed for given element
    Node:
    <input draggable="false" role="separator" style="cursor: col-resize;" class="c4p--datagrid__col-resizer-range" type="range" aria-label="Resize column" value="192">
    Fix any of the following:
      ARIA role separator is not allowed for given element

- Message: An explicit ARIA 'role' is not valid for <th> element within a ARIA role 'table' per the ARIA in HTML specification
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/div[1]/div[1]/table[1]/thead[1]/tr[1]/th[9]
      Snippet: <th colspan="1" scope="col" class="c4p--datagrid__resizableColumn" title="overview.languageTitle" aria-hidden="false" style="box-sizing: border-box; flex: 0 0 auto; min-width: 0px; width: 264px; position: relative;" role="columnheader">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/table_aria_descendants.html#%7B%22message%22%3A%22An%20explicit%20ARIA%20'role'%20is%20not%20valid%20for%20%3Cth%3E%20element%20within%20a%20ARIA%20role%20'table'%20per%20the%20ARIA%20in%20HTML%20specification%22%2C%22snippet%22%3A%22%3Cth%20colspan%3D%5C%221%5C%22%20scope%3D%5C%22col%5C%22%20class%3D%5C%22c4p--datagrid__resizableColumn%5C%22%20title%3D%5C%22overview.languageTitle%5C%22%20aria-hidden%3D%5C%22false%5C%22%20style%3D%5C%22box-sizing%3A%20border-box%3B%20flex%3A%200%200%20auto%3B%20min-width%3A%200px%3B%20width%3A%20264px%3B%20position%3A%20relative%3B%5C%22%20role%3D%5C%22columnheader%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22explicit_role%22%2C%22ruleId%22%3A%22table_aria_descendants%22%7D
- Message: An explicit ARIA 'role' is not valid for <tr> element within a ARIA role 'table' per the ARIA in HTML specification
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/div[1]/div[1]/table[1]/thead[1]/tr[1]
      Snippet: <tr class="c4p--datagrid__head c4p--datagrid__sticky" style="display: flex; flex: 1 0 auto; min-width: 0px; --c4p--datagrid--row-height: 0px; --c4p--datagrid--grid-height: -2px; --c4p--datagrid--header-height: 0px;" role="row">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/table_aria_descendants.html#%7B%22message%22%3A%22An%20explicit%20ARIA%20'role'%20is%20not%20valid%20for%20%3Ctr%3E%20element%20within%20a%20ARIA%20role%20'table'%20per%20the%20ARIA%20in%20HTML%20specification%22%2C%22snippet%22%3A%22%3Ctr%20class%3D%5C%22c4p--datagrid__head%20c4p--datagrid__sticky%5C%22%20style%3D%5C%22display%3A%20flex%3B%20flex%3A%201%200%20auto%3B%20min-width%3A%200px%3B%20--c4p--datagrid--row-height%3A%200px%3B%20--c4p--datagrid--grid-height%3A%20-2px%3B%20--c4p--datagrid--header-height%3A%200px%3B%5C%22%20role%3D%5C%22row%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22explicit_role%22%2C%22ruleId%22%3A%22table_aria_descendants%22%7D
- Message: An explicit ARIA 'role' is not valid for <td> element within a ARIA role 'table' per the ARIA in HTML specification
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/div[1]/div[1]/table[1]/tbody[1]/tr[9]/td[9]
      Snippet: <td class="c4p--datagrid__cell" style="box-sizing: border-box; flex: 0 0 auto; min-width: 0px; width: 264px;" role="cell">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/table_aria_descendants.html#%7B%22message%22%3A%22An%20explicit%20ARIA%20'role'%20is%20not%20valid%20for%20%3Ctd%3E%20element%20within%20a%20ARIA%20role%20'table'%20per%20the%20ARIA%20in%20HTML%20specification%22%2C%22snippet%22%3A%22%3Ctd%20class%3D%5C%22c4p--datagrid__cell%5C%22%20style%3D%5C%22box-sizing%3A%20border-box%3B%20flex%3A%200%200%20auto%3B%20min-width%3A%200px%3B%20width%3A%20264px%3B%5C%22%20role%3D%5C%22cell%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22explicit_role%22%2C%22ruleId%22%3A%22table_aria_descendants%22%7D
- Message: The ARIA role 'separator' is not valid for the element <input>
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/div[1]/div[1]/table[1]/thead[1]/tr[1]/th[9]/div[1]/input[1]
      Snippet: <input value="264" aria-label="Resize column" type="range" class="c4p--datagrid__col-resizer-range" style="cursor: col-resize;" role="separator" draggable="false">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/aria_role_valid.html#%7B%22message%22%3A%22The%20ARIA%20role%20'separator'%20is%20not%20valid%20for%20the%20element%20%3Cinput%3E%22%2C%22snippet%22%3A%22%3Cinput%20value%3D%5C%22264%5C%22%20aria-label%3D%5C%22Resize%20column%5C%22%20type%3D%5C%22range%5C%22%20class%3D%5C%22c4p--datagrid__col-resizer-range%5C%22%20style%3D%5C%22cursor%3A%20col-resize%3B%5C%22%20role%3D%5C%22separator%5C%22%20draggable%3D%5C%22false%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22aria_role_valid%22%7D
- Message: An element with ARIA role 'separator' does not have the required ARIA attribute(s): 'aria-valuenow'
      Level: violation
      XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/div[1]/div[1]/table[1]/thead[1]/tr[1]/th[9]/div[1]/input[1]
      Snippet: <input value="264" aria-label="Resize column" type="range" class="c4p--datagrid__col-resizer-range" style="cursor: col-resize;" role="separator" draggable="false">
      Help: https://able.ibm.com/rules/archives/latest/doc/en-US/aria_attribute_required.html#%7B%22message%22%3A%22An%20element%20with%20ARIA%20role%20'separator'%20does%20not%20have%20the%20required%20ARIA%20attribute(s)%3A%20'aria-valuenow'%22%2C%22snippet%22%3A%22%3Cinput%20value%3D%5C%22264%5C%22%20aria-label%3D%5C%22Resize%20column%5C%22%20type%3D%5C%22range%5C%22%20class%3D%5C%22c4p--datagrid__col-resizer-range%5C%22%20style%3D%5C%22cursor%3A%20col-resize%3B%5C%22%20role%3D%5C%22separator%5C%22%20draggable%3D%5C%22false%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22aria_attribute_required%22%7D

**Comment by elycheea** [comment]
@SeonyuK Can you take another since [this change](https://github.com/carbon-design-system/ibm-products/pull/3957) was merged?

**Comment by SeonyuK** [comment]
I can't update the IBM-Product version to 2.19.0.
After I update to 2.19.0, I get a Module not found error.
```
./node_modules/@carbon/ibm-products/lib/components/SteppedAnimatedMedia/SteppedAnimatedMedia.js
Module not found: Can't resolve 'lottie-web'
```

**Comment by SeonyuK** [comment]
@elycheea @davidmenendez 
Finally, I was able to update the version to 2.19.2.
I still have the same issue. I dug a little and found this. I think we should set 
`{...(role === 'separator' && { role })}` to `{...(role !== 'separator' && { role })}` (same as row and columnheader) so in case the input doesn't have a role as `separator` put the role but not for `separator`
I tested on your storybook and checked there is no role for input.
I think this will solve the accessibility test issue but not 100% sure yet..
![Screenshot 2024-01-04 at 2 04 55 PM](https://github.com/carbon-design-system/ibm-products/assets/144151347/7fbe6f2f-f5dd-4c6a-b26e-9485288a6980)
![Screenshot 2024-01-04 at 2 04 03 PM](https://github.com/carbon-design-system/ibm-products/assets/144151347/19146e7a-3911-43d9-bfe4-4f149863e25e)

**Comment by davidmenendez** [comment]
@SeonyuK the role related violations should be fixed in the next release.

**Comment by SeonyuK** [comment]
@davidmenendez  I updated the version to v 2.21.0 but still has one issue with batch action


>  - Message: Element "button" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true'
>       Level: violation
>       XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[2]/div[2]/button[2]
>       Snippet: <button type="button" class="cds--btn cds--btn--primary">
>       Help: https://able.ibm.com/rules/archives/latest/doc/en-US/aria_hidden_nontabbable.html#%7B%22message%22%3A%22Element%20%5C%22button%5C%22%20should%20not%20be%20focusable%20within%20the%20subtree%20of%20an%20element%20with%20an%20'aria-hidden'%20attribute%20with%20value%20'true'%22%2C%22snippet%22%3A%22%3Cbutton%20type%3D%5C%22button%5C%22%20class%3D%5C%22cds--btn%20cds--btn--primary%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22aria_hidden_nontabbable%22%7D
>     - Message: Element "button" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true'
>       Level: violation
>       XPath: /html[2]/body[1]/main[1]/div[1]/div[1]/div[1]/section[1]/div[2]/div[2]/button[1]
>       Snippet: <button type="button" class="cds--btn cds--btn--primary">
>       Help: https://able.ibm.com/rules/archives/latest/doc/en-US/aria_hidden_nontabbable.html#%7B%22message%22%3A%22Element%20%5C%22button%5C%22%20should%20not%20be%20focusable%20within%20the%20subtree%20of%20an%20element%20with%20an%20'aria-hidden'%20attribute%20with%20value%20'true'%22%2C%22snippet%22%3A%22%3Cbutton%20type%3D%5C%22button%5C%22%20class%3D%5C%22cds--btn%20cds--btn--primary%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22aria_hidden_nontabbable%22%7D
